tests/testthat/test-derive_param_confirmed_resp.R

```diff
@@ -0,0 +1,204 @@
+  "7",      "2020-04-01", "NE",
+  "9",      "2020-02-16", "PD",
+  "9",      "2020-03-06", "SD"
+) %>%
+  mutate(PARAMCD = "OVR") %>%
+  bind_rows(tibble::tribble(
+    ~USUBJID, ~ADTC,        ~AVALC,
+    "9",      "2020-02-16", "Y"
+  ) %>%
+    mutate(PARAMCD = "PD")) %>%
+  mutate(
+    ADT = lubridate::ymd(ADTC),
+    STUDYID = "XX1234"
+  )
+
+pd_date <- date_source(
+  dataset_name = "adrs",
+  date = ADT,
+  filter = PARAMCD == "PD"
+)
+
+# derive_param_confirmed_resp ----
+## derive_param_confirmed_resp Test 1: default confirmed response ----
+test_that("derive_param_confirmed_resp Test 1: default confirmed response", {
+  actual <-
+    derive_param_confirmed_resp(
+      adrs,
+      dataset_adsl = adsl,
+      filter_source = PARAMCD == "OVR",
+      source_pd = pd_date,
+      source_datasets = list(adrs = adrs),
+      ref_confirm = 28,
+      set_values_to = vars(
+        PARAMCD = "CRSP",
+        PARAM = "Confirmed Response by Investigator"
+      )
+    )
+
+  expected <- bind_rows(
+    adrs,
+    tibble::tribble(
+      ~USUBJID, ~ADTC,         ~AVALC, ~AVAL,
+      "1",      "2020-01-01",  "Y",    1,
+      "2",      NA_character_, "N",    0,
+      "3",      NA_character_, "N",    0,
+      "4",      NA_character_, "N",    0,
+      "5",      NA_character_, "N",    0,
+      "6",      NA_character_, "N",    0,
+      "7",      NA_character_, "N",    0,
+      "8",      NA_character_, "N",    0,
+      "9",      NA_character_, "N",    0
+    ) %>%
+      mutate(
+        ADT = lubridate::ymd(ADTC),
+        STUDYID = "XX1234",
+        PARAMCD = "CRSP",
+        PARAM = "Confirmed Response by Investigator"
+      )
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "PARAMCD", "ADT")
+  )
+})
+
+## derive_param_confirmed_resp Test 2: accept SD ----
+test_that("derive_param_confirmed_resp Test 2: accept SD", {
+  adrs_ext <- bind_rows(
+    filter(adrs, USUBJID != "7"),
+    tibble::tribble(
+      ~USUBJID, ~ADTC,        ~AVALC,
+      "7",      "2020-04-02", "ND"
+    ) %>%
+      mutate(
+        PARAMCD = "OVR",
+        ADT = lubridate::ymd(ADTC),
+        STUDYID = "XX1234"
+      )
+  )
+
+  actual <-
+    derive_param_confirmed_resp(
+      adrs_ext,
+      dataset_adsl = adsl,
+      filter_source = PARAMCD == "OVR",
+      source_pd = pd_date,
+      source_datasets = list(adrs = adrs),
+      ref_confirm = 14,
+      max_nr_ne = 0,
+      accept_sd = TRUE,
+      set_values_to = vars(
+        PARAMCD = "CRSP",
+        PARAM = "Confirmed Response by Investigator"
+      )
+    )
+
+  expected <- bind_rows(
+    adrs_ext,
+    tibble::tribble(
+      ~USUBJID, ~ADTC,         ~AVALC, ~AVAL,
+      "1",      "2020-01-01",  "Y",    1,
+      "2",      "2020-02-01",  "Y",    1,
+      "3",      "2019-11-12",  "Y",    1,
+      "4",      NA_character_, "N",    0,
+      "5",      "2020-01-01",  "Y",    1,
+      "6",      NA_character_, "N",    0,
```

Comment:
for USUBJID=6, interesting case and not thought about it before , we need to think more about this case. so if we consider 1 SD in between condition, then this can be Y? any thoughts?

Reply:
@bundfussr I think for the subject=6 after CR, we also have PR again. Its fine for now, I will dig more later